cylc/flow/tests/test_job_file.py

```diff
@@ -30,28 +33,428 @@
+def test_write_prelude_invalid_cylc_command(mocked_glbl_cfg):
+    job_conf = {
+        "batch_system_name": "background",
+        "host": "localhost",
+        "owner": "me"
+    }
+    mocked = mock.MagicMock()
+    mocked_glbl_cfg.return_value = mocked
+    mocked.get_host_item.return_value = 'cylc-testing'
+    with pytest.raises(ValueError) as ex:
+        with TemporaryFile(mode="w+") as handle:
+            JobFileWriter()._write_prelude(handle, job_conf)
+        assert("bad cylc executable" in str(ex))
+
+@mock.patch.dict(
+    "os.environ", {'CYLC_SUITE_DEF_PATH': 'cylc/suite/def/path'})
+@mock.patch("cylc.flow.job_file.get_remote_suite_run_dir")
+def test_write(mocked_get_remote_suite_run_dir):
+    """Test write function outputs jobscript file correctly"""
+    with TemporaryFile() as local_job_file_path:
+        local_job_file_path = str(local_job_file_path)
         job_conf = {
-            "batch_system_name": "background",
             "host": "localhost",
-            "owner": "me"
+            "owner": "me",
+            "task_id": "baa",
+            "suite_name": "farm_noises",
+            "work_d": "farm_noises/work_d",
+            "remote_suite_d": "remote/suite/dir",
+            "uuid_str": "neigh",
+            'environment': {'cow': '~/moo',
+                            'sheep': '~baa/baa',
+                            'duck': '~quack'},
+            "param_env_tmpl": {"param_env_tmpl_1": "moo",
+                               "param_env_tmpl_2": "baa"},
+            "job_d": "1/baa/01",
+            "try_num": 1,
+            "batch_system_name": "background",
+            "param_var": {"duck": "quack",
+                          "mouse": "squeak"},
+            "batch_submit_command_template": "woof",
+            "execution_time_limit": "moo",
+            "namespace_hierarchy": ["root", "baa", "moo"],
+            "dependencies": ['moo', 'neigh', 'quack'],
+            "init-script": "This is the init script",
+            "env-script": "This is the env script",
+            "err-script": "This is the err script",
+            "pre-script": "This is the pre script",
+            "script": "This is the script",
+            "post-script": "This is the post script",
+            "exit-script": "This is the exit script",
         }
-        mocked = mock.MagicMock()
-        mocked_glbl_cfg.return_value = mocked
-        mocked.get_host_item.return_value = 'cylc-testing'
-        with self.assertRaises(ValueError) as ex:
-            with TemporaryFile(mode="w+") as handle:
-                JobFileWriter()._write_prelude(handle, job_conf)
-        self.assertIn("bad cylc executable", str(ex.exception))
+        mocked_get_remote_suite_run_dir.return_value = "run/dir"
+        JobFileWriter().write(local_job_file_path, job_conf)
+
+        assert (os.path.exists(local_job_file_path))
+        size_of_file = os.stat(local_job_file_path).st_size
+        print(size_of_file)
+        assert(size_of_file == 1715)
```

Comment:
That's a somewhat strange way to test it!

This is effectively a "known good output" test, so we could just include the "good output", that way when the test breaks you get a diff.